platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,395 @@
+- CircleCI build logs
+- Slack notifications
+
+### Caveats
+
+### Security Concerns
+
+There are no new security concerns with a separated content validation process.
+
+### Privacy Concerns
+
+There are no new privacy concerns with a separated content validation process.
+
+### Open Questions and Risks
+
+#### Risks
+
+- Neither the current content validation nor the proposed content validation will **prevent** invalid content from making it to the live VA.gov site. But this proposal would prevent unrelated teams from being blocked by invalid content.
+
+#### Questions
+
+- What is the optimal frequency to run the scheduled job? Daily seems too slow. Every 10 minutes might be too fast for our builds. Hourly seems feasible. Is hourly acceptable? Should we be c concern about noise from a higher frequency?
+
+### Work Estimates
+
+The following estimates vary greatly depending on who's doing the work.
+
+1. Finish extracting the `content-build` repo out of the `vets-website` repo: ?
+2. Write a CircleCI scheduled job to run a script hourly on workdays: <1 day
+3. Write a script that builds only what is necessary for content validation: 1-2 days
+4. Update the script so the accessibility checks don't stop on failure: <1 day
+5. Update the script to generate a report of accessibility errors: <1 day
+6. Update the scheduled job to call the content validation script: <1 day
+
+### Alternatives
+
+#### Highest quality, but lowest speed
+
+If we could proactively validate CMS content before it's published, and sync the Drupal redirects with `vets-website`, we could eliminate a source of downstream broken links and/or accessibility errors, which would improve the experience for end users and/or reduce the number of frontend deploys that require manual intervention. The downside of this approach would be figuring out how to synchronize the Drupal redirects with `vets-website`, and/or adding friction to the content writers workflow.
+
+#### Fastest speed, but lowest quality
+
+Since the partial deploy already provides an avenue for invalid content to land on the live VA.gov site, we could remove all the content validation from the full `vets-website` build. That would reduce the burden on the content writers and/or reduce the number of deploys requiring manual intervention. However, that would likely mean end users experience broken links and/or accessibility errors more often.
+
+That would effectively be downgrading invalid content from an error to a warning.
+
+### Future Work
+
+- Delete the `accessibility` step of the `integration` stage from the `vets-website` Jenkins pipeline because that step will no longer be necessary in `vets-website`. It will no longer be necessary because the application accessibility checks are included in the tests run by the `e2e` step.
+
+### Revision History
+
+| Date         | Revisions Made | Author        |
+| ------------ | -------------- | ------------- |
+| May 26, 2020 | Initial draft  | Bill Fienberg |
```

Comment:
There have been many commits to this document since May 26, so this section is misleading. Consider deleting this since the revision history can be obtained from git?

Reply:
I see what you mean. Initially, I included more entries in the revision history, but I removed them per the following PR comment:

>I think you should only need to add revision history after the document has been approved and merged. - https://github.com/department-of-veterans-affairs/va.gov-team/pull/9658#discussion_r433466403